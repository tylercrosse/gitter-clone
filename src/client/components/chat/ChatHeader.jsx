import React from 'react';
import { Link } from 'react-router';

export const ChatHeader = ({ user }) => {
  const nav = user.loggedIn ? (
    <img
      className="avatar"
      src={'http://i.pravatar.cc/30?u=' + user.username} alt={user.username}
    />
  ) : (
    <Link to="/">Log In</Link>
  );
  return (
    <header className="chat-header">
      <Link to="/chat">Chat</Link>
      {nav}
    </header>
  );
};

export default ChatHeader;
