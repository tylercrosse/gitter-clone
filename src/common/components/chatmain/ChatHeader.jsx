/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

export const ChatHeader = ({ signOut, user, pathname }) => {
  const userInfo = user.loggedIn && (
    <div className="chat-user-avatar" >
      <img
      onClick={() => signOut()}
      className="avatar"
      src={'http://i.pravatar.cc/30?u=' + user.name} alt={user.name}
      />
    </div>
  );
  return (
    <header className="chat-header">
      <div className="chat-header-avatar">
        <img
          src={'http://i.pravatar.cc/32?u=' + pathname} alt={pathname}
        />
      </div>
      <Link
        className="chat-header-title"
        to={'/' + pathname}
      >
        {pathname}
      </Link>
      {userInfo}
    </header>
  );
};

ChatHeader.propTypes = {
  user: PropTypes.object.isRequired,
  pathname: PropTypes.string.isRequired,
  signOut: PropTypes.func.isRequired,
};

export default ChatHeader;
