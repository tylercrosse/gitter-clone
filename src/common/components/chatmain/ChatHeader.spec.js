import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ChatHeader from './ChatHeader';

const setup = (propOverrides) => {
  const props = Object.assign({
    user: {
      loggedIn: true,
      name: 'dan'
    }
  }, propOverrides);

  const component = <ChatHeader {...props} />;

  return {
    props,
    component
  };
};

describe('<ChatHeader />', () => {
  describe('logged in', () => {
    it('should render correctly', () => {
      const { component } = setup();
      const renderedComponent = renderer.create(component);
      const tree = renderedComponent.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('logged out', () => {
    it('should render correctly', () => {
      const { component } = setup({user: {loggedIn: false}});
      const renderedComponent = renderer.create(component);
      const tree = renderedComponent.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
