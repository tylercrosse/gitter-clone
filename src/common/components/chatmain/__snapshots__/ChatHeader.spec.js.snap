// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> logged in should render correctly 1`] = `
<header
  className="chat-header"
>
  <div
    className="chat-header-avatar"
  >
    <img
      alt={undefined}
      src="http://i.pravatar.cc/32?u=undefined"
    />
  </div>
  <a
    className="chat-header-title"
    onClick={[Function]}
    style={Object {}}
  />
  <div
    className="chat-user-avatar"
  >
    <img
      alt="dan"
      className="avatar"
      onClick={[Function]}
      src="http://i.pravatar.cc/30?u=dan"
    />
  </div>
</header>
`;

exports[`<ChatHeader /> logged out should render correctly 1`] = `
<header
  className="chat-header"
>
  <div
    className="chat-header-avatar"
  >
    <img
      alt={undefined}
      src="http://i.pravatar.cc/32?u=undefined"
    />
  </div>
  <a
    className="chat-header-title"
    onClick={[Function]}
    style={Object {}}
  />
</header>
`;
